keyvault/client/src/main/java/com/azure/keyvault/KeyvaultCredentialsPolicy.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.credentials.ServiceClientCredentials;
+import com.azure.common.http.*;
+import com.azure.common.http.policy.HttpPipelinePolicy;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+final class KeyvaultCredentialsPolicy implements HttpPipelinePolicy {
```

Comment:
I need to update my spec about pipeline policies - should they be public API or not. I also need to update my spec to specify where policies should go, because it shouldn't be in the root package most probably.

Reply:
if we let developers make their HttpPipelines from scratch, then having this public should be useful for them.
One could argue that if they are making their own HttpPipeline from scratch, then they will make their own CredentialsPolicy from scratch too, but I lean towards having it public if we let them build their own HttpPipeline, credentials policy will not need to be re-invented in terms of logic.